src/components/DotIndicator/src/dot-indicator.css

```diff
@@ -0,0 +1,21 @@
+.denhaag-dot-indicator {
+  position: relative;
+  display: inline-flex;
+  flex-shrink: 0;
+  vertical-align: middle;
+}
+
+.denhaag-dot-indicator__dot {
+  position: absolute;
+  background-color: var(--denhaag-dot-indicator-background-color);
+  width: var(--denhaag-dot-indicator-size);
+  height: var(--denhaag-dot-indicator-size);
+  top: 0;
+  right: 0;
+  border-radius: 50%;
+  border: var(--denhaag-dot-indicator-border);
+}
+
+.denhaag-dot-indicator--overlap-rectangle .denhaag-dot-indicator__dot  {
+  transform: scale(1) translate(50%, -50%);
+}
```

Comment:
I've decided to only add a modifier for the rectangle overlap. Only the rectangle overlap needs specific styling, even though (from the React API POV), there's a circle and rectangle overlap. If we also specify the `--overlap-circle` modifier it will would have no styling attached to it.

Reply:
Deciding to stick with the bem notation for consistency